projects/js-packages/publicize-components/src/hooks/use-publicize-config/index.js

```diff
@@ -1,17 +1,26 @@
 import {
 	getJetpackExtensionAvailability,
 	isUpgradable,
+	getJetpackData,
 } from '@automattic/jetpack-shared-extension-utils';
 import { useSelect, useDispatch } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
 
 const republicizeFeatureName = 'republicize';
 
+/**
+ * Hook that provides various elements of Publicize configuration,
+ * whether it's enabled, and whether resharing is available.
+ *
+ * @returns { object } The various flags and togglePublicizeFeature,
+ * for toggling support for the current post.
+ */
 export default function usePublicizeConfig() {
 	const { togglePublicizeFeature } = useDispatch( 'jetpack/publicize' );
-	const { available: isRePublicizeFeatureAvailable } = getJetpackExtensionAvailability(
-		republicizeFeatureName
-	);
+	const sharesData = getJetpackData()?.social?.sharesData ?? {};
+	const isShareLimitEnabled = sharesData.is_share_limit_enabled;
+	const isRePublicizeFeatureAvailable =
+		getJetpackExtensionAvailability( republicizeFeatureName )?.available || isShareLimitEnabled;
```

Comment:
So, the feature flag for `isShareLimitEnabled` controls on what sites we make resharing free. When we make it free without feature flags, what would we need to do?

Something like this?
```
		\Jetpack_Gutenberg::set_extension_available( 'republicize' );
```

Reply:
Eventually we'd remove the check, but we could just have the server side check for the feature flag return that the feature is enabled.